cpp/src/arrow/array/array_struct_test.cc

```diff
@@ -196,6 +196,22 @@ TEST(StructArray, Validate) {
   ASSERT_RAISES(Invalid, arr->ValidateFull());
 }
 
+TEST(StructArray, ValidateFullNullable) {
+  auto type = struct_({field("a", int32(), /*nullable =*/false),
+                       field("b", utf8(), /*nullable =*/false),
+                       field("c", list(boolean()), /*nullable =*/false)});
+
+  auto struct_arr = std::static_pointer_cast<StructArray>(ArrayFromJSON(
+      type, R"([[1, "a", [null, false]], [null, "bc", []], [2, null, null]])"));
+
+  auto struct_arr_nonull = std::static_pointer_cast<StructArray>(ArrayFromJSON(
+      type, R"([[1, "a", [true, false]], [6, "bc", []], [2, "bcj", [true, true]]])"));
```

Comment:
Note that there's no need to cast the result, since ValidateFull is a method of the base class.